File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // prompt user for input
    string text = get_string("Text: ");
    int LENGTH = strlen(text);

    int words = 0;
    int letters = 0;
    int sentences = 0;

    char digit;

    // doing everything here prevents iterating the same string on 3 functions
    for (int i = 0; i <= LENGTH; i++)
    {
        // prevent uppercase and reevaluating the same char
        digit = tolower(text[i]);

        if (isalpha(digit))
        {
            letters++;
        }
        else if (isspace(digit) || digit == '\0')
        {
            words++;
        }
        else if (digit == '.' || digit == '?' || digit == '!')
        {
            sentences++;
        }
    }

    // note that if multiplying by '100' will alter our result since float * int
    float avg_letters = letters * 100.0 / words;
    float avg_sentences = sentences * 100.0 / words;

    int index = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+");
    }
    else if (index < 1)
    {
        printf("Before Grade 1");
    }
    else
    {
        printf("Grade %i", index);
    }

    printf("\n");
}

File: readability.py
from cs50 import get_string


# Main program section
def main():
    # Prompt user for input
    text = get_string("Text: ")

    words = 0
    letters = 0
    sentences = 0

    for char in text:
        # Is alphabetical
        if char.isalpha():
            letters += 1
        # Is space
        elif char.isspace():
            words += 1
        # Is end of sentence
        elif char in [".", "?", "!"]:
            sentences += 1

    avg_letters = letters * 100 / words
    avg_sentences = sentences * 100 / words

    # Print Coleman-Liau index rounded to nearest integer
    index = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# If running from this file
if __name__ == "__main__":
    # Code execution begins here"
    main()